File: Backend/Nodejs/src/libs/s3client.ts
import { S3Client } from "@aws-sdk/client-s3";
import { configS3 } from "../config/aws";
import multerS3 from "multer-s3";

const s3Client = new S3Client(configS3);

export const s3StorageImg = ({ userType }: { userType: string }) =>
  multerS3({
    s3: s3Client,
    bucket: process.env.AWS_BUCKET_NAME as string,
    contentType: multerS3.AUTO_CONTENT_TYPE,
    metadata: function (req, file, cb) {
      cb(null, { fieldName: file.fieldname });
    },
    key: function (req, file, cb) {
      const fileName = "Fotos/" + Date.now() + userType +
        file.originalname;
      cb(null, fileName);
    },
  });

File: Backend/Python/src/libs/s3client.py
from src.config.aws import session
from decouple import config
from src.utils.current_time import date_now

AWS_BUCKET_NAME = config("AWS_BUCKET_NAME")

s3_client = session.client("s3")
location = s3_client.get_bucket_location(Bucket=AWS_BUCKET_NAME)
region = location["LocationConstraint"]
url = "https://%s.s3.%s.amazonaws.com/" % (AWS_BUCKET_NAME, region)
print(url)


def upload_file(user, file, img=True):
    try:
        dir = "Fotos/"
        if not img:
            dir = "Canciones/"
        filename = dir + str(date_now()) + user + file.filename
        s3_client.put_object(
            Body=file,
            Bucket=AWS_BUCKET_NAME,
            Key=filename,
            ContentType=file.mimetype,
        )
        return url + filename
    except Exception as e:
        print(e)
        return ""
